### DISCUSSION TITLE
Problem with using Websend in Tasmota rules

### DISCUSSION THREAD

**Discussion by TKBeckum** [arendst/Tasmota#21693] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
I want to send a webhook to a Homey Bridge. The command that is expected by the homey bridge is:
webhook.homey.app/secret-key/Lamps?tag=On
When I use this line in a webbrowser the homey bridge reacts as expected and turn the lights on.
However, when I use the following command in Tasmota (in rules or as a command in the console):
Websend[webhook.homey.app]/sevret-key/Lamps?tag=On]
The console says: RESULT = {"WebSend":"Done"}, but the homey bridge doesn't turn the lamps on.
What is the problem?

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ x] Device used (e.g., Sonoff Basic): esp32 wroom
- [x ] Tasmota binary firmware version number used: TASMOTA-on-ESP32-master downloaded 12-06-2024
  - [x ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
16:43:46.287 RSL: RESULT = {"NAME":"ESP32-DevKit","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1216,0,1,1,1,0,0,0,0,1,1,4704,1,1,0,0,1],"FLAG":0,"BASE":1}
16:43:46.507 RSL: RESULT = {"Module":{"1":"ESP32-DevKit"}}
16:43:46.765 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"1216":"AM2301"},"GPIO24":{"0":"None"},"GPIO25":{"0":"None"},"GPIO26":{"0":"None"},"GPIO27":{"0":"None"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"4768":"ADC Light1"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here: (Ichnges the real homey key into 'secret-key'
16:45:53.293 RSL: RESULT = {"Rule1":{"State":"OFF","Once":"ON","StopOnError":"OFF","Length":116,"Free":395,"Rules":"ON Time#Minute<=900 DO backlog rule2 0; rule3 0; ENDON ON Time#Minute>900 DO backlog rule2 1; rule3 0; rule1 0 ENDON"}}
16:45:53.517 RSL: RESULT = {"Rule2":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":430,"Free":81,"Rules":"ON Time#Minute<=900 DO backlog rule1 1; rule3 0; rule2 0 ENDON ON ANALOG#Illuminance1>250 DO backlog WebSend[webhook.homey.app/secret-key/Leeslampen?tag=Aan]; rule3 1; rule2 0 ENDON ON ANALOG#Illuminance1<15 DO WebSend[webhook.homey.app/secret-key/Leeslampen?tag=Aan]; ENDON  ON Time#Minute>1380 DO backlog WebSend [webhook.homey.app/secret-key/Leeslampen?tag=Uit]; rule1 1; rule2 0 ENDON"}}
16:45:53.717 RSL: RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":203,"Free":308,"Rules":"ON Time#Minute<=900 DO backlog rule1 1; rule2 0; rule3 0; ENDON ON ANALOG#Illuminance1<5 DO backlog WebSend [https://webhook.homey.app/secret-key/Leeslampen?tag=Uit]; rule1 1; rule3 0 ENDON"}}

```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
16:48:00.875 RSL: STATUS = {"Status":{"Module":1,"DeviceName":"Esp32-Lux-Kamer","FriendlyName":["Tasmota"],"Topic":"tasmota_58AE58","ButtonTopic":"0","Power":"0","PowerLock":"0","PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
16:48:00.886 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Software reset CPU","Uptime":"0T01:05:51","StartupUTC":"2024-06-24T13:42:09","Sleep":50,"CfgHolder":4617,"BootCount":35,"BCResetTime":"2024-06-12T10:30:45","SaveCount":660}}
16:48:00.898 RSL: STATUS2 = {"StatusFWR":{"Version":"14.1.0(release-tasmota32)","BuildDateTime":"2024-06-03T11:56:19","Core":"3_0_0","SDK":"5.1.4.240529","CpuFrequency":160,"Hardware":"ESP32-D0WD-V3 v3.1","CR":"426/699"}}
16:48:00.909 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Sitecom437EFA",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000081","00006000","00004000","00000000"]}}
16:48:00.935 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":2033,"Free":846,"Heap":146,"StackLowMark":3,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":"DIO","Features":["0809","9F9AD7DF","0015A001","B7F7BFCF","05DA9BC4","E0360DC7","480840D2","20200000","D4BC482D","810A80B1","00000014"],"Drivers":"1,2,!3,!4,!5,7,!8,9,10,11,12,!14,!16,!17,!20,!21,!24,26,!27,29,!34,!35,38,50,52,!59,!60,62,!63,!66,!67,!68,!73,82,!86,!87,!88,!121","Sensors":"1,2,3,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,31,34,37,39,40,42,43,45,51,52,55,56,58,59,64,66,67,74,85,92,95,98,103,105,109,127","I2CDriver":"7,8,9,10,11,12,13,14,15,17,18,20,24,29,31,36,41,42,44,46,48,58,62,65,69,76,77,82"}}
16:48:00.949 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-58AE58-3672","IPAddress":"192.168.0.125","Gateway":"192.168.0.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.0.1","DNSServer2":"0.0.0.0","Mac":"A0:B7:65:58:AE:58","IP6Global":"","IP6Local":"fe80::a2b7:65ff:fe58:ae58%st1","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"192.168.0.1","DNSServer2":"0.0.0.0","Mac":"00:00:00:00:00:00","IP6Global":"","IP6Local":""},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":16.0}}
16:48:00.960 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_58AE58","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
16:48:00.970 RSL: STATUS7 = {"StatusTIM":{"UTC":"2024-06-24T14:48:00Z","Local":"2024-06-24T16:48:00","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":99,"Sunrise":"05:47","Sunset":"21:57"}}
16:48:00.987 RSL: STATUS10 = {"StatusSNS":{"Time":"2024-06-24T16:48:00","ANALOG":{"Illuminance1":116},"AM2301":{"Temperature":22.9,"Humidity":59.0,"DewPoint":14.4},"TempUnit":"C"}}
16:48:01.001 RSL: STATUS11 = {"StatusSTS":{"Time":"2024-06-24T16:48:00","Uptime":"0T01:05:51","UptimeSec":3951,"Heap":144,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":4,"Objects":46},"Wifi":{"AP":1,"SSId":"Sitecom437EFA","BSSId":"64:D1:A3:43:7E:FA","Channel":11,"Mode":"HT20","RSSI":84,"Signal":-58,"LinkCount":1,"Downtime":"0T00:00:04"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
16:50:03.747 CMD: WebSend[webhook.homey.app/secret-key/Leeslampen?tag=Aan]
16:50:03.748 SRC: WebConsole from 192.168.0.110
16:50:03.749 CMD: Grp 0, Cmd 'WEBSEND', Idx 1, Len 63, Pld -99, Data '[webhook.homey.app/secret-key/Leeslampen?tag=Aan]'
16:50:03.849 RSL: RESULT = {"WebSend":"Done"}

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Use websend to send a webhook to a homey bridge

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
the correct reaction of the homey bridge (=turning Lamps on)

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
Again: sending the command via webbrowser works, so it is not a problem of the homey bridge.

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by barbudor** [comment, marked as answer]
If Tasmota says "done", it means it done its job
Doesn't mean that what you asked it to do is correct regarding the other end's server

```
Websend[webhook.homey.app]/sevret-key/Lamps?tag=On]
```
Here, it's likely there is an extraneous `]` which could be the problem (unless it is a typo when posting)

When addressing another webserver (non tasmota device), WebQuery is most likely to be better than WebSend

Do you have a sample request that works ? 
Such as a curl command line ?

From thatpage : https://support.homey.app/hc/en-us/articles/4410240765586-Using-Homey-Logic-in-Flows
It looks like the correct URL is:
```
https://webhook.homey.app/5edb9248470f0276f4b63151/my_event?tag=my_tag
```
I would use:
```
WebQuery  https://webhook.homey.app/5edb9248470f0276f4b63151/my_event?tag=my_tag GET
```

**Reply by TKBeckum** [reply]
Thanks for your reply to my question. Unfortunately, it didn't bring a solution nearer (maybe because of my ignorance).
First of all, the [ ] around the url are necessary. Without them Tasmota says "WebSend":"Wrong parameters".
Second, adding GET to the webhook doesn't turn the lamps on either.
I also tried a webhook without a tag, but that didn't work either.
I hope you have other suggestions. Thanks in advance

**Reply by sfromis** [reply]
Basically, just do **not** use `WebSend`, it works better with `WebQuery` where you can simply have an URL as a parameter, instead of the weirdness of the `WebSend` syntax. If changing to the right syntax for `WebSend`, you could not have it use `https` instead of `http`, pretty much invalidating usage of Homey through `WebSend`.

**Reply by TKBeckum** [reply]
Thanks very much. Your suggestion is the right one! When I use in the console:
Webquery https://webhook.homey.app/secret-key/Test GET
Homey turns on the lamp. It also works when specifying a tag. Now I will try to use this command in a rule. I keep you updated.
So far,so good! Thanks again for your help!

**Reply by barbudor** [reply]
You should read every words in one's reponse

**Reply by TKBeckum** [reply]
Yes, you are absolutely right. My apologies! I don't know why I didn't see you already suggested the solution in your response. Frustration because I couldn't find a solution to my problem didn't help me to carefully study your response. Sorry again!
